wakuv2/waku.go

```diff
@@ -430,53 +429,24 @@ func (w *Waku) discoverAndConnectPeers() error {
 				continue
 			}
 
-			go w.identifyAndConnect(w.ctx, w.cfg.LightClient, *peerInfo)
+			go w.connect(w.ctx, *peerInfo)
 		}
 	}
 
 	return nil
 }
 
-func (w *Waku) identifyAndConnect(ctx context.Context, isLightClient bool, peerInfo peer.AddrInfo) {
-	ctx, cancel := context.WithTimeout(ctx, 7*time.Second)
+func (w *Waku) connect(ctx context.Context, peerInfo peer.AddrInfo) {
+	ctx, cancel := context.WithTimeout(ctx, dialTimeout)
 	defer cancel()
 
+	// Connection will be prunned eventually by the connection manager if needed
+	// Connect waits until Identify protocol has been executed
 	err := w.node.Host().Connect(ctx, peerInfo)
```

Comment:
One thing i am wondering here is instead of using host's connect API, it may be better to use WakuNode's AddDiscoveredPeer API which internally connects to the peer as well(based on option specified).
It also takes care of storing relevant metadata for the peer in WakuPeerStore which helps with other capabilities. 

https://github.com/waku-org/go-waku/blob/8115ec701354befb4e790944ffeba312a372c50e/waku/v2/node/wakunode2.go#L701

Reply:
Sounds good. Will implement that change